lib/query-interface.js

```diff
@@ -495,18 +495,20 @@ QueryInterface.prototype.insert = function(instance, tableName, values, options)
   });
 };
 
-QueryInterface.prototype.upsert = function(tableName, values, updateValues, model, options) {
+QueryInterface.prototype.upsert = function(tableName, valuesByField, updateValues, model, options) {
```

Comment:
Well, I'm not sure what usually happens in your code base :) What I know is that when a PK attribute has a `.field` the passed in values object will be keyed on the `.field` name rather than the attribute name. The previous code seemed to assume that the values were keyed by attribute name.